ecommers/lib/core/repositories/repository_base.dart

```diff
@@ -0,0 +1,31 @@
+import 'dart:convert';
+import 'package:ecommers/core/models/index.dart';
+import 'package:ecommers/shared/dependency_service.dart';
+
+class RepositoryBase<T extends ItemBase> {
+  final String filterFieldForItem;
+  final String repositoryKey;
+  
+  RepositoryBase({
+    this.filterFieldForItem,
+    this.repositoryKey,
+  });
+
+  Future edit(T item) async {
+    await cacheDatabase.updateByEqualsFilter(
```

Comment:
can be without async and just return(please check whole file)

Reply:
fuxed